runtime/jsMain/src/kotlinx/benchmark/NodeJsEngineSupport.kt

```diff
@@ -3,14 +3,19 @@ package kotlinx.benchmark
 import kotlin.time.DurationUnit
 import kotlin.time.toDuration
 
-internal external fun require(module: String): dynamic
+@JsModule("fs")
```

Comment:
This makes sense since the docs say kx-benchmark only supports `nodejs()`.

However, I see the library also has a `D8EngineSupport`, so doesn't this change mean a require or import of `fs` will always be performed eagerly? Unlike it was done before - lazily?

Reply:
(Lucky me) Right now, D8 is not supported for K/JS: https://github.com/Kotlin/kotlinx-benchmark/blob/da3e2bbad26f5b531af8c9c39431ca1933ed525b/plugin/main/src/kotlinx/benchmark/gradle/JsEngineExecTasks.kt#L45
And for K/WasmJs, we use a different executor at runtime.

Reply:
BTW, is it possible easily to support D8 for K/JS (for JB personal use)?

Reply:
Sure, we can try.

Reply:
@fzhinkin might have better info, but my understanding is you'd need [KT-20679](https://youtrack.jetbrains.com/issue/KT-20679).

The fact is:
1. annotating externals with `@JsModule` translates into eager resolution
2. custom lazy loading means you'd have to know when CommonJS is used to `require()`, or when ESM is used to use  dynamic `import()`

Apart from D8, in the future people might want to run benchmarks in a standard browser environment, but yeah, we seem to be limited now.

Might workaround limitation 2 by having a build-time constant inject, like

```kt
// Injected by a compiler plugin?
const val IMPORT: String = "require('...')" // or import('...')

// At use site
js(IMPORT).writeFileSync(path, text)
```

Reply:
> Sure, we can try.

Cool, especially if we can do that with the same executor as `Wasm` do. We need it to compare `JS` and `WasmJS` performance. But it is not priority thing anyway.

Reply:
Filed https://github.com/Kotlin/kotlinx-benchmark/issues/293 to track the proposal.

Reply:
> but my understanding is you'd need [KT-20679](https://youtrack.jetbrains.com/issue/KT-20679).

We know the underlying runtime before generating the code, so we may try generating some part of the executor with appropriate import/require statements along with the benchmarks itself.

Reply:
Yeah that sounds much better compared to working around it with custom lazy loading.